testing/unit_tests/test_patch_filter.py

```diff
@@ -0,0 +1,91 @@
+import os
+import pytest
+import numpy as np
+import torch.backends.cudnn as cudnn
+from torch.utils.data import DataLoader
+
+from ivadomed.loader.bids_dataframe import BidsDataframe
+from ivadomed import utils as imed_utils
+from ivadomed.loader import utils as imed_loader_utils, loader as imed_loader
+from testing.unit_tests.t_utils import create_tmp_dir,  __data_testing_dir__, __tmp_dir__, download_data_testing_test_files, path_repo_root
+from testing.common_testing_util import remove_tmp_dir
+
+cudnn.benchmark = True
+
+GPU_ID = 0
+BATCH_SIZE = 1
+
+
+def setup_function():
+    create_tmp_dir()
+
+
+def _cmpt_slice(ds_loader):
+    cmpt_label = {0: 0, 1: 0}
+    for i, batch in enumerate(ds_loader):
+        for gt in batch['gt']:
+            # TODO: multi label
+            if np.any(gt.numpy()):
+                cmpt_label[1] += 1
+            else:
+                cmpt_label[0] += 1
+    print(cmpt_label)
+    return cmpt_label[0], cmpt_label[1]
+
+
+@pytest.mark.parametrize('transforms_dict', [{"CenterCrop": {"size": [128, 128], "applied_to": ["im", "gt"]}}])
+@pytest.mark.parametrize('train_lst', [['sub-rat3_ses-01_sample-data9_SEM.png']])
+@pytest.mark.parametrize('target_lst', [["_seg-axon-manual", "_seg-myelin-manual"]])
+@pytest.mark.parametrize('patch_filter_params', [
+    {"filter_empty_mask": False, "filter_empty_input":  True},
+    {"filter_empty_mask": True, "filter_empty_input": True}])
+@pytest.mark.parametrize('dataset_type', ["training", "testing"])
```

Comment:
FYI, in the near future, we will most likely start the pytest-cases migration to avoid these gigantic parametrize block which is very hard to keep track.